### DISCUSSION TITLE
Send MIDI event

### DISCUSSION THREAD

**Discussion by DreadPirateLynx** [sezanzeb/input-remapper#406] [Ideas]
Would it be possible to add the function to remap button presses to MIDI events?

**Comment by jonasBoss** [comment]
we are using [UInput](https://python-evdev.readthedocs.io/en/latest/tutorial.html#injecting-input) to remap the events. I don't know how MIDI works. If it relies on evedev under the hood, it should be possible. 
If you can research how MIDI works and maybe link some resources I can make a better assessment.

**Reply by DreadPirateLynx** [reply]
I'll take a look around; I'm still trying to figure out how it works in Linux myself

**Reply by hjri** [reply]
MIDI generally works over alsa/pulse/pipewire/jack and it works by sending/receiving events like note, control change, program change, system messages etc.

Generally MIDI is used to produce music, i.e. by sending the said events to the synthesizer, but MIDI is also useful in reverse by having a simple midi controller, like this one:  
![image](https://github.com/sezanzeb/input-remapper/assets/3514245/6cff3911-f341-420e-beef-178234f31bfb)  
typically those would be used for transport control in a DAW, adjust channels in a mixer or adjust filters/effects etc, but it's extremely flexible, I've set up mine to control PipeWire/Pulseaudio volumes https://ebin.club/hj-blog/convenience-audio-setup-part-2-hardware-knobs/
